frontend/src/hooks/useUserSession.js

```diff
@@ -0,0 +1,27 @@
+import axios from "axios";
+
+import { useExceptionHandler } from "../hooks/useExceptionHandler.jsx";
+import { useAlertStore } from "../store/alert-store";
+const useUserSession = () => {
+  const handleException = useExceptionHandler();
+  const { setAlertDetails } = useAlertStore();
+  return async () => {
+    try {
+      const requestOptions = {
+        method: "GET",
+        url: "/api/v1/session",
+      };
+      const res = await axios(requestOptions);
+      return res.data;
+    } catch (error) {
+      if (error?.response?.statusText === "Payment Required") {
+        handleException(error, "Error while getting session");
```

Comment:
@athul-rs In order to display the error message in the popup, we need to pass the `handleException` function inside `setAlertDetails`. Please refer line 22.

**For more context:**
**handleException**: is a custom hook that returns an object containing all necessary fields: title, content, and type.

**setAlertDetails**: is a setter function used to update the global state variable "alertDetails". Whenever this state is updated, the popup message will be displayed.

Reply:
For this added conditional statements the pop-up messages are not required, thus the reason those are not added.